File: examples/python/src/class/MyStack.py
class MyStack:
    def __init__(self, size):
        self.maxSize = size
        self.items = []
    
    def push(self, data):
        if self.items.__len__ < self.maxSize
            self.items.append(data)

    def pop(self):
        return self.items.pop()
    
    def peek(self):
        if self.items:
            return self.items[-1]
    
    def isEmpty(self):
        return self.items == []

    def isFull(self):
        return (self.items.__len__() == (self.maxSize - 1))

if __name__ == '__main__':
    s = MyStack(5)
    s.push(10)
    s.push(20)
    s.push(30)
    s.push(40)
    s.push(50)

    while (s.isEmpty() == False):
        print(s.pop())
        print(' ')

    print ('end')

File: examples/c/src/class/MyStack.cpp
#include <iostream>       // std::cout
#include <stack>          // std::stack
#include <vector>         // std::vector
#include <deque>          // std::deque
#include <stdio.h>

class MyStack
{
private:
    int maxSize = -1;
    int top = -1;
    long stackArray[0];
public:
    MyStack(int s);

    void push(long j)
    {
        stackArray[++top] = j;
    }

    long pop() 
    {
      return stackArray[top--];
    }
    
    long peek() {
      return stackArray[top];
    }
    
    bool isEmpty() {
      return (top == -1);
    }
    
    bool isFull() {
      return (top == maxSize);
    }

};

MyStack::MyStack(int s)
{
    maxSize = s;
    stackArray[maxSize];
    top = -1;
}

int main() 
{
      MyStack theStack(5); 
      theStack.push(10);
      theStack.push(20);
      theStack.push(30);
      theStack.push(40);
      theStack.push(50);
      
      printf ("Print stack values:\n");

      while (!theStack.isEmpty()) {
         long value = theStack.pop();
         printf ("%ld\n", value);
         printf ("------\n");
      }
      printf ("End.\n");

      return 0;
}